src/prefect/flow_engine.py

```diff
@@ -658,6 +658,27 @@ def initialize_run(self):
                 },
             )
 
+            if flow_run_ctx := FlowRunContext.get():
+                if traceparent := flow_run_ctx.flow_run.labels.get(
+                    "__OTEL_TRACEPARENT"
+                ):
+                    propagate.get_global_textmap().inject({"traceparent": traceparent})
+                else:
+                    carrier = {}
+                    propagate.get_global_textmap().inject(
+                        carrier, context=trace.set_span_in_context(span)
+                    )
+                    if carrier.get("traceparent"):
+                        self.logger.info(
+                            f"Setting traceparent {carrier['traceparent']} for flow run {self.flow_run.name!r}"
+                        )
+                        self.client.update_flow_run_labels(
+                            flow_run_id=self.flow_run.id,
+                            labels={"__OTEL_TRACEPARENT": carrier["traceparent"]},
+                        )
+
+            self._span = span
```

Comment:
maybe we could pull bespoke key names like `__OTEL_TRACEPARENT` and `traceparent` up the top as `SOME_CONSTANT_VALUE`?